src/ralph/models/xapi/fields/context.py

```diff
@@ -0,0 +1,62 @@
+"""Common xAPI context field definitions"""
+
+from ipaddress import IPv4Address
+from pathlib import Path
+from typing import Optional
+
+from pydantic import Field, constr
+
+from ..config import BaseModelWithConfig
+from ..constants import (
+    XAPI_EXTENSION_ACCEPT_LANGUAGE,
+    XAPI_EXTENSION_AGENT,
+    XAPI_EXTENSION_COURSE_ID,
+    XAPI_EXTENSION_COURSE_USER_TAGS,
+    XAPI_EXTENSION_HOST,
+    XAPI_EXTENSION_IP,
+    XAPI_EXTENSION_ORG_ID,
+    XAPI_EXTENSION_PATH,
+    XAPI_EXTENSION_SESSION,
+)
+
+
+class ContextExtensionsField(BaseModelWithConfig):
+    """Represents the `context.extensions` xAPI field.
+
+    WARNING: It doesn't include the optional `referer` field.
+
+    Attributes:
+        ip (IPv4Address): Consists of the public IPv4 address of the user. (Optional)
+        agent (str): Consists of the `User-Agent` HTTP request header. (Optional)
+        host (str): Consists of the hostname of the server. (Optional)
+        accept_language (str): Consists of the `Accept-Language` HTTP request header. (Optional)
+        course_id (str): Consists of the unique identifier for the visited course page. (Optional)
```

Comment:
Do you have an example course ID?

Reply:
Good point) Thanks) Added the example and regex validation for course_id)